src/client/components/vision/camera/view_model.ts

```diff
@@ -0,0 +1,282 @@
+  }
+
+  private image = createTransformer((image: ImageModel): Mesh => {
+    const mesh =  new Mesh(this.quadGeometry, this.imageMaterial(image))
+    mesh.scale.y = -1 // We need to flip Y to fix the image coordinates
+    return mesh
+  })
+
+  @computed
+  private get quadGeometry(): BufferGeometry {
+    return new PlaneBufferGeometry(2, 2)
+  }
+
+  // This shader must be stored separately as a computed field as if we make it new each time,
+  // it will get compiled each time which is quite expensive.
+  @computed
+  private get imageBasicMaterial() {
+    return new MeshBasicMaterial({
+      depthTest: false,
+      depthWrite: false,
+    })
+  }
+
+  private imageMaterial = createTransformer((image: ImageModel) => {
+    const mat = this.imageBasicMaterial
+
+    // TODO: This is wrong and not very reactive however the alternative is recompiling the shader every time
+    // you have a better idea?
+    mat.map = this.imageTexture(image)
+    return mat
+  })
+
+  // not computed as otherwise it will update the textures too frequently
+  private imageTexture = createTransformer((image: ImageModel) => {
+    switch (image.format) {
+      case fourcc('GRBG'):
+      case fourcc('RGGB'):
+      case fourcc('GBRG'):
+      case fourcc('BGGR'):
+        return this.bayerTexture(image).texture
+      case fourcc('RGB3'):
+        return this.rgbTexture(image)
+      case fourcc('YUYV'):
+        return this.yuyvTexture(image)
+      case fourcc('GREY'):
+      case fourcc('GRAY'):
+        return this.grayTexture(image)
+      default:
+        throw Error('Unsupported image format')
+    }
+  })
+
+  // This shader must be stored separately as a computed field as if we make it new each time,
+  // it will get compiled each time which is quite expensive.
+  @computed
+  private get bayerShader() {
+    return new RawShaderMaterial({
+      vertexShader: String(bayerVertexShader),
+      fragmentShader: String(bayerFragmentShader),
+      uniforms: {
+        sourceSize: { value: new Vector4() },
+        firstRed: { value: new Vector2() },
+        image: { type: 't' },
+      },
+      depthTest: false,
+      depthWrite: false,
+    })
+  }
+
+  private bayerTexture = createTransformer((image: ImageModel) => {
+    let firstRed
+    switch (image.format) {
+      case fourcc('GRBG'):
+        firstRed = [1, 0]
+        break
+      case fourcc('RGGB'):
+        firstRed = [0, 0]
+        break
+      case fourcc('GBRG'):
+        firstRed = [0, 1]
+        break
+      case fourcc('BGGR'):
+        firstRed = [1, 1]
+        break
+    }
+
+    const { width, height } = image
+    const renderTarget = new WebGLRenderTarget(width, height)
+    renderTarget.depthBuffer = false
+    renderTarget.stencilBuffer = false
+    const scene = new Scene()
+    const material = this.bayerShader
+
+    // TODO: This is wrong and not very reactive however the alternative is recompiling the shader every time
+    // you have a better idea?
```

Comment:
Why is it recompiling the shader each time exactly?

Reply:
Every time you make a material, it compiles the shader needed for that material. Previously the Bayer material was made here but because of that, it recompiled the shader every time.